File: day10/CountWords.java
import java.util.Scanner;
import java.util.HashMap;

public class CountWords {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        // Prompt the user to enter a sentence
        System.out.print("Enter a sentence: ");
        String input = scanner.nextLine().toLowerCase(); // convert to lowercase to ignore case

        // Split the sentence into individual words by spaces
        String[] words = input.split("\\s+"); // "\\s+" matches one or more spaces

        // Create a HashMap to store each word and its count
        HashMap<String, Integer> wordCounts = new HashMap<>();

        // Loop through each word
        for (String word : words) {
            // If the word is already in the map, increment its count
            // Otherwise, add it with a count of 1
            wordCounts.put(word, wordCounts.getOrDefault(word, 0) + 1);
        }

        // Print out each word and how many times it appeared
        for (String word : wordCounts.keySet()) {
            System.out.println(word + ": " + wordCounts.get(word));
        }

        scanner.close(); // Close the scanner when done
    }
}

File: day10/countWords.cpp
#include <iostream>
#include <string>
#include <sstream>
#include <map>
#include <algorithm>

int main() {
    std::string input;
    std::map<std::string, int> wordCounts;

    // Prompt the user for input
    std::cout << "Enter a sentence: ";
    std::getline(std::cin, input);

    // Convert input to lowercase
    std::transform(input.begin(), input.end(), input.begin(), ::tolower);

    // Use a stringstream to split the input into words
    std::stringstream ss(input);
    std::string word;

    // Extract words and count them
    while (ss >> word) {
        wordCounts[word]++;
    }

    // Output each word and its frequency
    for (const auto& pair : wordCounts) {
        std::cout << pair.first << ": " << pair.second << std::endl;
    }

    return 0;
}
